src/fs/visit.rs

```diff
@@ -14,7 +14,7 @@ use super::{ModFile, SmashPath, RejectionReason};
 /// Discover every file in a directory and its sub-directories.  
 /// Files starting with a period are filtered out, and only the files with relevant regions are kept.  
 /// This signifies that if your goal is to simply get all the files, this is not the method to use.
-pub fn discovery<Arc: ArcLookup>(arc: &Arc, path: &PathBuf, accepted: &mut HashMap<Hash40, ModFile>, rejected: &mut Vec<(PathBuf, RejectionReason)>) {
+pub fn discovery<Arc: ArcLookup>(arc: &Arc, path: &PathBuf, accepted: &mut HashMap<Hash40, ModFile>, rejected: &mut Vec<(PathBuf, RejectionReason)>, stream: &mut HashMap<Hash40, ModFile>) {
```

Comment:
This function signature is getting messy. Consider returning a struct named `DiscoveryResult` that has accepted, rejected, and stream as fields. Unless you intend on accepted/rejected/stream not starting empty I'd prefer that quite a bit.